ifrs17/Import/ImportStorage.ipynb

```diff
@@ -224,53 +224,52 @@
+                "\n                                                 || secondaryScopeAtInceptionYear.Contains(iv.DataNode)).ToArrayAsync();",
+                "\n        }",
                 "\n",
                 "\n        if(allImportScopesNotAtInceptionYear.Any()) {",
-                "\n            PreviousPeriodPartition = (await querySource.Query<PartitionByReportingNodeAndPeriod>().Where(p => p.ReportingNode == args.ReportingNode &&",
-                "\n                                                                                                           p.Year == PreviousReportingPeriod.Year &&",
-                "\n                                                                                                           p.Month == PreviousReportingPeriod.Month &&",
-                "\n                                                                                                           p.Scenario == null).ToArrayAsync()).Single().Id;",
+                "\n            PreviousPeriodPartition = (await querySource.Query<PartitionByReportingNodeAndPeriod>()",
+                "\n                                        .Where(p => p.ReportingNode == args.ReportingNode && p.Year == PreviousReportingPeriod.Year ",
+                "\n                                                 && p.Month == PreviousReportingPeriod.Month && p.Scenario == null).ToArrayAsync()).Single().Id;",
                 "\n            ",
                 "\n            await querySource.Partition.SetAsync<PartitionByReportingNodeAndPeriod>(PreviousPeriodPartition);",
                 "\n        ",
                 "\n            //Perform queries to previous Period",
                 "\n            openingRawVariables = (await querySource.Query<RawVariable>()",
                 "\n                                   .Where(rv => rv.Partition == PreviousPeriodPartition && rv.AocType == AocTypes.CL)",
-                "\n                                   .Where(v => primaryScope.Contains(v.DataNode))",
-                "\n                                   .ToArrayAsync())",
-                "\n                .Select(rv => rv with {AocType = AocTypes.BOP, Novelty = Novelties.I, Values = rv.Values.Skip(MonthInAYear).ToArray(), Partition = TargetPartition});",
+                "\n                                   .Where(v => primaryScope.Contains(v.DataNode)).ToArrayAsync())",
+                "\n                                   .Select(rv => rv with {AocType = AocTypes.BOP, Novelty = Novelties.I, ",
+                "\n                                                 Values = rv.Values.Skip(MonthInAYear).ToArray(), Partition = TargetPartition});",
                 "\n            ",
-                "\n            openingIfrsVariables = (await querySource.Query<IfrsVariable>()",
+                "\n            openingIfrsVariables = openingIfrsVariables.Union((await querySource.Query<IfrsVariable>()",
```

Comment:
why Union? a dataNode is either at inception year or not. There's no overlap between the two sets.

Reply:
yes, this is correct! However, the union is more robust and will prevent that changes to the inception logic cause a data duplication. I suggest to keep the Union. What do you propose to do?